packages/core/src/modules/message-pickup/protocol/MessagePickupProtocolOptions.ts

```diff
@@ -1,12 +1,33 @@
 import type { AgentMessage } from '../../../agent/AgentMessage'
 import type { ConnectionRecord } from '../../connections'
+import type { QueuedMessage } from '../storage'
 
 export interface PickupMessagesProtocolOptions {
   connectionRecord: ConnectionRecord
   recipientKey?: string
   batchSize?: number
 }
 
+export interface DeliverMessagesProtocolOptions {
+  connectionRecord: ConnectionRecord
+  messages?: QueuedMessage[]
+  recipientKey?: string
```

Comment:
I think here the same, even though the RFC describes what should be used as the value for `recipientKey`, I'd prefer to make the variable more explicit, or make it a class instance and transform it last minute when adding it to the message